ionoscloud/resource_k8s_cluster.go

```diff
@@ -621,18 +621,23 @@ func setK8sClusterData(d *schema.ResourceData, cluster *ionoscloud.KubernetesClu
 }
 
 func k8sClusterReady(ctx context.Context, client *ionoscloud.APIClient, d *schema.ResourceData) (bool, error) {
-	subjectCluster, apiResponse, err := client.KubernetesApi.K8sFindByClusterId(ctx, d.Id()).Execute()
+	resource, apiResponse, err := client.KubernetesApi.K8sFindByClusterId(ctx, d.Id()).Execute()
 	logApiRequestTime(apiResponse)
-
 	if err != nil {
 		return true, fmt.Errorf("error checking k8s cluster status: %w", err)
 	}
-	return *subjectCluster.Metadata.State == "ACTIVE", nil
+	if resource.Metadata == nil || resource.Metadata.State == nil {
```

Comment:
I think if the Metadata is nil it will try to check also the Metadata.State if you use OR